src/rest/src/main/java/org/geoserver/rest/AbstractResource.java

```diff
@@ -92,7 +96,7 @@ protected DataFormat getFormatGet() {
         if ( ext == null ) {
             ext = (String) getRequest().getAttributes().get( "type" );
         }
-        if ( ext == null ) {
+        if ( ext == null && includeFileExtension) {
```

Comment:
@NielsCharlier extra space after opening bracket. Yes, you did not add it, but this is a great time to fix it.